tests/HypothesisClient/RequestConstraint.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace tests\eLife\HypothesisClient;
+
+use GuzzleHttp\Psr7\Request;
```

Comment:
`Psr\Http\Message\MessageInterface` would be better, though `str()` doesn't normalize.

Reply:
In the sense that we could also compare `Response`s? Comparing class plus result of `str()` would be enough?

Reply:
I'm going to merge this for now and we can scrutinise this more at a later time. Thanks for your assistance.